projects/keycloak/CertificateUtilsProviderFuzzer.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 the cncf-fuzzing authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////
+import com.code_intelligence.jazzer.api.FuzzedDataProvider;
+import java.io.ByteArrayInputStream;
+import java.security.KeyPair;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Date;
+import org.keycloak.common.crypto.CertificateUtilsProvider;
+import org.keycloak.common.util.KeyUtils;
+import org.keycloak.crypto.def.BCCertificateUtilsProvider;
+import org.keycloak.crypto.elytron.ElytronCertificateUtils;
+import org.keycloak.crypto.fips.BCFIPSCertificateUtilsProvider;
+
+/**
+ * This fuzzer targets the methods in different
+ * Certificate Utils Provider implementation classes
+ * in the crypto package.
+ */
+public class CertificateUtilsProviderFuzzer {
+  public static void fuzzerTestOneInput(FuzzedDataProvider data) {
+    // Initialise base certificate related object
+    CertificateUtilsProvider provider = null;
+    X509Certificate cert = null;
+    KeyPair keyPair = null;
+
+    try {
+      // Randomly create a certificate utils provider instance
+      switch (data.consumeInt()) {
+        case 1:
+          provider = new BCCertificateUtilsProvider();
```

Comment:
Does the fuzzer need to generate a provider on every iteration, or can a global provider be instantiated and reused?

Reply:
fixed.